webapp/src/js/app.js

```diff
@@ -134,21 +134,22 @@ _.templateSettings = {
     contacts: 'can_view_contacts',
     analytics: 'can_view_analytics',
     reports: 'can_view_reports',
+    'reports.edit': 'can_update_reports'
   };
 
   var getRequiredPermission = function(route) {
     var baseRoute = route.split('.')[0];
-    return ROUTE_PERMISSIONS[baseRoute];
+    return _.uniq(_.compact([ROUTE_PERMISSIONS[baseRoute], ROUTE_PERMISSIONS[route]]));
```

Comment:
I wonder if it would be cleaner for ROUTE_PERMISSIONS to have an array of permissions so we don't have to do this manipulation every time.

Reply:
Sure! I didn't want to introduce a need to duplicate base route permissions, but it's little effort to check.